python/chemiscope/structures/_ase.py

```diff
@@ -320,25 +351,41 @@ def extract_vectors_from_ase(frames, key="forces", **kwargs):
 
     vectors = []
 
+    # set shape parameters globally if they are all given
+    globs = {}
+    if "radius" in kwargs:
+        globs["base_radius"] = kwargs.pop("radius")
+        if "head_radius_scale" in kwargs:
+            globs["head_radius"] = globs["base_radius"] * kwargs.pop(
+                "head_radius_scale"
+            )
+        if "head_length_scale" in kwargs:
+            globs["head_length"] = globs["base_radius"] * kwargs.pop(
+                "head_length_scale"
+            )
+
     for f in frames:
-        if key not in f.arrays:
-            raise IndexError(f"Key {key} not found in `Atoms.arrays`")
-        values = f.arrays[key]
+        values, target = _extract_key_from_ase(f, key, target)
+        if target == "structure":
+            values = values.reshape(1, -1)
         if len(values.shape) != 2 or values.shape[1] != 3:
             raise ValueError(
                 f"Property array {key} has not the shape of a list of 3-vectors"
             )
 
         # makes a list of arrows to visualize the property
-        vectors.append([arrow_from_vector(v, **kwargs) for v in values])
+        vectors = vectors + [arrow_from_vector(v, **kwargs) for v in values]
 
-    return vectors
+    if target == "atom":
+        return {"kind": "arrow", "parameters": {"global": globs, "atom": vectors}}
+    else:
+        return {"kind": "arrow", "parameters": {"global": globs, "structure": vectors}}
 
 
-def extract_tensors_from_ase(frames, key="tensor", **kwargs):
+def ase_tensors_to_ellipsoids(frames, key="tensor", target=None, **kwargs):
```

Comment:
```suggestion
def ase_tensors_to_ellipsoids(frames, key, target=None, **kwargs):
```

I would not set a default, since it will depend a lot on the application.

Reply:
```suggestion
def ase_tensors_to_ellipsoids(frames, key="tensor", target=None, scale=1.0, force_positive=False):
```

Making the parameters explicit (and documented, even if just to point to the documentation for `ellipsoid_from_tensor`) will be nicer.